src/ip_notifier.cpp

```diff
@@ -74,10 +222,29 @@ namespace libtorrent
 	void ip_change_notifier::async_wait(std::function<void(error_code const&)> cb)
 	{
 #if defined TORRENT_BUILD_SIMULATOR
-		TORRENT_UNUSED(cb);
+		// TODO: simulator support
+		cb(make_error_code(boost::system::errc::not_supported));
 #elif TORRENT_USE_NETLINK
+		using namespace std::placeholders;
 		m_socket.async_receive(boost::asio::buffer(m_buf)
 			, std::bind(&ip_change_notifier::on_notify, this, _1, _2, cb));
+#elif TORRENT_USE_SYSTEMCONFIGURATION
+		m_cb = std::move(cb);
```

Comment:
I looked at this, and I can't convert a lambda with capture to a function pointer. And I coded this in a way that once the subscription is done, you can only change the "callback member", not safe to use from multiple clients.